src/app/storage/FabricTableImpl.h

```diff
@@ -0,0 +1,158 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/TypeTraits.h>
+
+namespace chip {
+namespace app {
+namespace Storage {
+
+/**
+ * @brief Implementation of a storage accessor in nonvolatile storage of a templatized table that stores by fabric index.
+ *        This class does not actually hold the entries, but rather acts as a wrapper/accessor around the storage layer,
+ *        reading entries from the storage pointed to by calling SetEndpoint.
+ *
+ * FabricTableImpl is an implementation that allows to store arbitrary entities using PersistentStorageDelegate.
+ * It handles the storage of entities by their StorageId and EnpointID over multiple fabrics.
+ */
+template <class StorageId, class StorageData, size_t kIteratorsMax>
+class FabricTableImpl
+{
+    using TableEntry    = Data::TableEntry<StorageId, StorageData>;
+    using EntryIterator = CommonIterator<TableEntry>;
+    using EntryIndex    = Data::EntryIndex;
+
+public:
+    virtual ~FabricTableImpl() { Finish(); };
+
+    CHIP_ERROR Init(PersistentStorageDelegate * storage);
+    void Finish();
+
+    // Entry count
+    /**
+     * @brief Get the total number of stored entries for the entire endpoint
+     * @param entry_count[out] the count of entries
+     * @return CHIP_ERROR, CHIP_NO_ERROR if successful or if the Fabric was not found, specific CHIP_ERROR otherwise
+     */
+    CHIP_ERROR GetEndpointEntryCount(uint8_t & entry_count);
+
+    /**
+     * @brief Get the total number of stored entries for the specified fabric
```

Comment:
```suggestion
     * @brief Get the total number of stored entries for the specified fabric on the currently selected endpoint.
```
or something like that, right?